applications/nrf_desktop/src/modules/ble_qos.c

```diff
@@ -0,0 +1,689 @@
+{
+	struct config_fetch_event *fetch_event;
+	struct params_blacklist *qos_blacklist;
+
+	fetch_event = new_config_fetch_event(sizeof(struct params_blacklist));
+	__ASSERT_NO_MSG(fetch_event != NULL);
+	qos_blacklist = (struct params_blacklist *) fetch_event->dyndata.data;
+	memset(qos_blacklist, 0, sizeof(struct params_blacklist));
+
+	qos_blacklist->wifi_chn_bitmask = chmap_filter_wifi_blacklist_get();
+
+	return fetch_event;
+}
+
+static void update_parameters(
+	struct params_ble *p_qos_ble_params,
+	struct params_wifi *p_qos_wifi_params)
+{
+	struct chmap_filter_params *p_params;
+
+	p_params = k_malloc(sizeof(struct chmap_filter_params));
+	if (p_params == NULL) {
+		LOG_ERR("OOM error");
+		return;
+	}
+
+	chmap_filter_params_get(mp_chmap_inst, p_params);
+
+	if (p_qos_ble_params != NULL) {
+		p_params->min_channel_count =
+			p_qos_ble_params->min_channel_count;
+		p_params->ble_weight_crc_ok =
+			p_qos_ble_params->weight_crc_ok;
+		p_params->ble_weight_crc_error =
+			p_qos_ble_params->weight_crc_error;
+		p_params->ble_block_threshold =
+			p_qos_ble_params->ble_block_threshold;
+		p_params->eval_max_count =
+			p_qos_ble_params->eval_max_count;
+		p_params->eval_duration =
+			p_qos_ble_params->eval_duration;
+		p_params->maintenance_sample_count =
+			p_qos_ble_params->sample_count_min;
+		p_params->eval_keepout_duration =
+			p_qos_ble_params->eval_keepout_duration;
+		p_params->eval_success_threshold =
+			p_qos_ble_params->eval_success_threshold;
+	}
+	if (p_qos_wifi_params != NULL) {
+		p_params->wifi_rating_inc =
+			p_qos_wifi_params->wifi_rating_inc;
+		p_params->wifi_present_threshold =
+			p_qos_wifi_params->wifi_present_threshold;
+		p_params->wifi_active_threshold =
+			p_qos_wifi_params->wifi_active_threshold;
+	}
+
+	k_mutex_lock(&m_data_access_mutex, K_FOREVER);
+	if (m_new_params != NULL) {
+		k_free(m_new_params);
+	}
+	m_new_params = p_params;
+	k_mutex_unlock(&m_data_access_mutex);
+}
+
+static void update_blacklist(struct params_blacklist *p_blacklist)
+{
+	atomic_set(&m_new_blacklist, p_blacklist->wifi_chn_bitmask);
+}
+
+static int hci_chmap_cmd_send(const u8_t *p_chmap)
+{
+	struct bt_hci_cp_le_set_host_chan_classif *cp;
+	struct net_buf *buf;
+
+	__ASSERT_NO_MSG(p_chmap != NULL);
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_SET_HOST_CHAN_CLASSIF,
+				sizeof(*cp));
+	if (!buf) {
+		return -1;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	if (!cp) {
+		return -1;
+	}
+
+	memcpy(cp->ch_map, p_chmap, sizeof(cp->ch_map));
+	__ASSERT_NO_MSG(sizeof(cp->ch_map) == 5);
+
+	return bt_hci_cmd_send_sync(
+			BT_HCI_OP_LE_SET_HOST_CHAN_CLASSIF,
+			buf,
+			NULL);
+}
+
+static bool on_vs_evt(struct net_buf_simple *buf)
+{
+	u8_t *p_subevent_code;
+	hci_vs_evt_qos_conn_event_report_t *p_evt;
+
+	p_subevent_code = net_buf_simple_pull_mem(
+						buf,
+						sizeof(*p_subevent_code));
+
+	switch (*p_subevent_code) {
+	case HCI_VS_SUBEVENT_CODE_QOS_CONN_EVENT_REPORT:
+	{
```

Comment:
Do we need brackets here ?

Reply:
No, remnant from a variable declaration that has been removed.